livre/eq/Config.cpp

```diff
@@ -60,6 +97,65 @@ class Config::Impl
         , dataFrameRange( INVALID_FRAME_RANGE )
     {}
 
+    void gatherHistogram( const Histogram& histogram, const float area, const uint32_t currentId )
+    {
+        const uint32_t oldestId = histogramQueue.empty() ? -1u :
+                                  histogramQueue.back().id;
+
+        // If we get a very old frame skip it
+        if( oldestId != -1u && currentId < oldestId )
+            return;
+
+        const ViewHistogram viewHistogram( histogram, area, currentId );
+        auto it = histogramQueue.begin();
+        while( it != histogramQueue.end( ))
+        {
+            auto& data = *it;
+            bool dataMerged = false;
+
+            if( currentId == data.id )
+            {
+                dataMerged = true;
+                data += viewHistogram;
+            }
+            else if( currentId > data.id )
+            {
+                dataMerged = true;
+                it = histogramQueue.emplace( it, viewHistogram );
+            }
+
+            if( (*it).isComplete( )) // Send histogram & remove all old ones
+            {
+#ifdef LIVRE_USE_ZEROEQ
+                communicator->publishHistogram( (*it ).histogram );
+#endif
+                histogramQueue.erase( it, histogramQueue.end( ));
```

Comment:
There may be previous ones, still accumulating according to the latency. Therefore no.